script.js

```diff
@@ -53,81 +67,137 @@ my_widget_script =
+              moles.val((amount.val() / fw.val()).toFixed(nFixed))
+              my_widget_script.change_record_value()
+            }
+          } else {
+            var equiv = $('input[name^=equivalents]', tr);
+            var moles1 = $('#the_form input[name=moles1_number]');
+            if (fw.val()) {
+              moles.val((amount.val() / fw.val()).toFixed(nFixed));
+              equiv.val((amount.val() / fw.val() / moles1.val()).toFixed(nFixed))
+            }
           }
-        });
+        }),
+
+            $('#the_form input[name^=fw]').on('change', function() {
+              var tr = $(this).closest('tr');
+              console.log('In fw change handler '+ tr.attr('class'))
+              var is_initial_row = tr.attr('class') == 'initialRow';
+              var fw = $('input[name^=fw]', tr);
+              var amount = $('input[name^=amount]', tr);
+              var moles = $('input[name^=moles]', tr);
+              if (is_initial_row) {
+                if (amount.val() && !moles.val()) {
+                  moles.val((amount.val() / fw.val()).toFixed(nFixed))
+                } else if (moles.val() && !amount.val()) {
+                  amount.val((moles.val() * fw.val()).toFixed(nFixed))
+                }
+                if (amount.val() && moles.val()) {
+                  my_widget_script.enable_all_records()
+                }
+              } else {
+                var equiv = $('input[name^=equivalents]', tr);
+                var moles1 = $('#the_form input[name=moles1_number]');
+                if (amount.val() && !equiv.val()) {
+                  moles.val((amount.val() / fw.val()).toFixed(nFixed))
+                  equiv.val((amount.val() / fw.val() / moles1.val()).toFixed(nFixed))
+                } else if ((equiv.val() && !amount.val()) || (equiv.val() && amount.val())) {
+                  moles.val((moles1.val() * equiv.val()).toFixed(nFixed));
+                  var value = $('input[name^=moles]', tr);
+                  amount.val((value.val() * fw.val()).toFixed(nFixed))
+                }
+              }
+            }),
+
+            $('#the_form input[name^=moles]').on('change', function() {
+              var tr = $(this).closest('tr');
+              console.log('In moles change handler '+ tr.attr('class'))
+              var is_initial_row = tr.attr('class') == 'initialRow';
+              var fw = $('input[name^=fw]', tr);
+              var amount = $('input[name^=amount]', tr);
+              var moles = $('input[name^=moles]', tr);
+              if (is_initial_row) {
+                if (fw.val() && !amount.val()) {
+                  amount.val((moles.val() * fw.val()).toFixed(nFixed))
+                  my_widget_script.enable_all_records()
+                } else if (fw.val() && amount.val() && moles.val()) {
+                  amount.val((moles.val() * fw.val()).toFixed(nFixed))
+                  my_widget_script.change_record_value()
+                }
+              }
```

Comment:
so we just silently do nothing if moles is updated on non-initial rows? I suppose that's not terrible, but if so shouldn't we explicitly disable the non-moles `<input>`s in the UI?

Reply:
yes, no action is needed if mole value are changed for non-initial rows as it is not a use case scenario for us. feel there is not a need to disable in UI for consistent behaviour.

Reply:
Not sure what you mean by consistent behaviour. It would be consistent with disabling equiv in the initial row to disable moles in non-initial rows.